### PR TITLE
skopeo: drop support for ostree

### PR DESCRIPTION
**Pull request by giuseppe**
drop support for the ostree backend.  The only known user for the
backend is the atomic CLI tool that is not maintained anymore
upstream.

Closes: https://bugzilla.redhat.com/show_bug.cgi?id=1766404

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by mtrmac** [comment]
Sure, if you say so – but tests are failing.

**Comment by mtrmac** [comment]
… but, well, “not maintained” is not the same as “is not used”; just today I received some questions around `atomic trust`+`atomic sign`.

Is the transition plan / branching / RHEL impact known in detail?

**Comment by giuseppe** [comment]
> … but, well, “not maintained” is not the same as “is not used”; just today I received some questions around `atomic trust`+`atomic sign`.
> 
> Is the transition plan / branching / RHEL impact known in detail?

I've talked with Dan about it and while atomic+skopeo is supported, we are not going to use a newer Skopeo version in RHEL 7.

**Comment by vrothberg** [comment]
We could use this occasion and bundle it with the skopeo-sync PR and bump Skopeo to v1.0.0 :birthday:

**Comment by rhatdan** [comment]
Lets create a RHEL7 branch, incase we need to have a skopeo CVE Fix and then you can merge the changes.

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ export GOPROXY=https://proxy.golang.org
 
 ifeq ($(shell uname),Darwin)
 PREFIX ?= ${DESTDIR}/usr/local
-DARWIN_BUILD_TAG=containers_image_ostree_stub
+DARWIN_BUILD_TAG=containers_image
```
Comment (mtrmac) [inline Makefile]: What is this tag supposed to do?

**Comment by vrothberg** [comment]
@giuseppe can you cherry-pick the commit fromhttps://github.com/containers/skopeo/pull/758 and include it in this PR? This will bundle dropping ostree with the storage update.

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@ export GOPROXY=https://proxy.golang.org
 
 ifeq ($(shell uname),Darwin)
 PREFIX ?= ${DESTDIR}/usr/local
-DARWIN_BUILD_TAG=containers_image_ostree_stub
+DARWIN_BUILD_TAG=
```
Comment (mtrmac) [inline Makefile]: (Non-blocking: This could be removed completely.)

**Comment by mtrmac** [comment]
> @giuseppe can you cherry-pick the commit fromhttps://github.com/[/pull/758](https://github.com/containers/skopeo/pull/758) and include it in this PR? This will bundle dropping ostree with the storage update.

We could also just merge this (removing the user) before #758 (removing the implementation), keeping the Dependabot PR ~unmodified. Is there a strong reason to bundle them? (I don’t feel at all strongly about this.)

**Comment by vrothberg** [comment]
> > @giuseppe can you cherry-pick the commit fromhttps://github.com/[/pull/758](https://github.com/containers/skopeo/pull/758) and include it in this PR? This will bundle dropping ostree with the storage update.
> 
> We could also just merge this (removing the user) before #758 (removing the implementation), keeping the Dependabot PR ~unmodified. Is there a strong reason to bundle them? (I don’t feel at all strongly about this.)

Sure, sounds good to me.

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**